### DISCUSSION TITLE
Welcome to Racer Discussions!

### DISCUSSION THREAD

**Discussion by StuckAtPrototype** [StuckAtPrototype/Racer#7] [Announcements]
## 👋 Hi! Thanks for checking out the project! 
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

Thanks so much for the amazing support from everyone! I never expected this to blow up this much, so again thank you. Lets make fun stuff together. 

Here is my plan for this project as of right now:
- [ ] Add a BOM for all necessary parts and where to get them 
- [ ] Expand the Readme to include more details on how to get started 

I will happily expand this list as questions arise. I am happy to add as much detail as needed for everyone to be able to enjoy this project! 

Stay fun, 
StuckAtPrototype

**Comment by rudra1601** [comment]
Hello! @StuckAtPrototype Thank you for sharing this amazing project ❤️.

I am still confused about one thing, why did you have to use a Neural Network for Color Detection??
Wasn't it possible to use a simple Euclidean Distance Formula for detecting the color. It would save a lot of computation and reduce a lot of overhead.

**Reply by StuckAtPrototype** [reply]
Hi @RUDisCOOL! Thats fair, I should put this answer in the readme :) 

a TLDR answer: I wanted to learn more about neural nets and I needed to account for changing lighting conditions. As many have pointed out, I could have converted the color space, etc -- probably. 

The long answer: Initially I tried using a bracketing method (i.e. a simple switch case with a range), that fell on its face very quickly. During the day, it would work, at night it would not, etc. Despite the fact that I put in lights under the car to try and reduce the effect from an external light on the color. With a neural net, I was able to teach the network to recognize the "singular" (in quotes since the hue of the color shifts so much dependent on the lighting conditions) color despite shifting lighting conditions. Its an overkill, but it works! Plus it was super fun to learn and experiment with. In the end, if its not fun then what is it? :) 

I hope this answers the question!
